src/Core/Services/MetadataProviders/SqlMetadataProvider.cs

```diff
@@ -1240,12 +1240,14 @@ public static bool IsGraphQLReservedName(Entity entity, string databaseColumnNam
             string schemaName,
             string tableName)
         {
-            DataTable? dataTable = EntitiesDataSet.Tables[tableName];
+            // MySQL does not use schema
+            string schemaAndTableName = _databaseType is DatabaseType.MySQL ? tableName : $"{schemaName}.{tableName}";
```

Comment:
The string `{queryPrefix}{SqlQueryBuilder.QuoteIdentifier(tableName)}` obtained in FillSchemaForTableAsync seems to be the right way to generate the complete schema name/table name, why dont we move that into a helper and reuse it here and in `FillSchemaForTableAsync`?
That should handle MySQL automatically, since it checks for string.IsNullOrEmpty...

Reply:
modified the helper function `GetTablePrefix` into `GetTableNameWithPrefix` and included the database name as part of the string we return, so that if we ever have a single meta data provider per data source name we will de-conflict collisions between schema and table names that are the same but in different data bases.